### PR TITLE
Rename `PayPalNativeCheckout` to `PayPalNativePayments`

### PR DESCRIPTION
**Pull request by jcnoriega**
Thank you for your contribution to PayPal. 

> Before submitting this PR, note that we cannot accept language translation PRs. We have a dedicated localization team to provide the translations. If there is an error in a specific translation, you may open an issue and we will escalate it to the localization team.

### Summary of changes

Rename `PayPalNativeCheckout` to `PayPalNativePayments` which includes:
- Updating `build.gradle` references
- Updating github actions
- Changing package name
- Updating docs

[Ticket DTNOR-782](https://paypal.atlassian.net/browse/DTNOR-782)
 ### Checklist

 - [x] Added a changelog entry

- @jcnoriega

### CODE REVIEW

File: docs/PayPalNativeCheckout/README.md [hunk 1]
```diff
@@ -17,28 +17,28 @@ Follow the steps in [Get Started](https://developer.paypal.com/api/rest/#link-ge
...
-In order to integrate PayPalNative checkout, you will need:
+In order to integrate PayPal native payments, you will need:
 
 1. A PayPal client ID and corresponding secret. This is to generate an access token that will allow you to create payment tokens, capture funds and authorize customers to place orders.
 2. Setting a return URL.
 
 A return URL is required for redirecting users back to the app after authenticating. Please reference our [developer documentation](https://developer.paypal.com/docs/business/native-checkout/android/) to create said url and also to learn about how to create a new PayPal application as well.
 
-### 3. Initiate PayPal Native Checkout
+### 3. Initiate PayPal native checkout
```
Comment (sshropshire) [inline docs/PayPalNativeCheckout/README.md]: 🧶: Step 5 has different casing for "native checkout"

File: Demo/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
     package="com.paypal.android">
 
     <!-- We add this so that we can test on API 21, because NXO doesn't support below 23 (for now)  -->
-    <uses-sdk tools:overrideLibrary="com.paypal.android.checkout"/>
+    <uses-sdk tools:overrideLibrary="com.paypal.android.nativepayments"/>
```
Comment (sshropshire) [inline Demo/src/main/AndroidManifest.xml]: 🧶: should the package name be `com.paypal.android.paypalnativepayments` to match the artifact id?

**Review by sshropshire — APPROVED** [review]
🌊 Couple nits. LGTM

File: Demo/src/main/AndroidManifest.xml (hunk 2 above)
Comment (jcnoriega) [reply Demo/src/main/AndroidManifest.xml]: we could for consistency, but isnt a bit redudant to have like `paypal` 2 times under the package?

Comment (sshropshire) [reply Demo/src/main/AndroidManifest.xml]: Definitely redundant, but it should be fine. The convention across modules seems to be the full artifact ID in the last component of the package name, and the `com.paypal` prefix is supposed to callout that these libraries are manufactured by the same people who own [paypal.com](https://www.youtube.com/watch?v=dQw4w9WgXcQ).

Comment (sshropshire) [reply Demo/src/main/AndroidManifest.xml]: DON'T CLICK THE LINK! 😭

Comment (jcnoriega) [reply Demo/src/main/AndroidManifest.xml]: hahahaha lol

Yeah it makes sense
i'll change it

**Review by mattwylder — APPROVED** [review]
LGTM. Is there a company-wide change from "Native Checkout" to "Native Payments"?

**Comment by jcnoriega** [comment]
@mattwylder not that i know, but we did run this name changes with the whole team, not just the mobile crew.

**Pull request merged**